File: trading_app/lib/features/auth/domain/auth_models.dart
class AppUser {
  const AppUser({
    required this.id,
    required this.phone,
    required this.nickname,
    required this.role,
  });

  final String id;
  final String phone;
  final String nickname;
  final String role;

  factory AppUser.fromJson(Map<String, dynamic> json) {
    return AppUser(
      id: json['id'] as String,
      phone: json['phone'] as String,
      nickname: json['nickname'] as String,
      role: json['role'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'id': id,
      'phone': phone,
      'nickname': nickname,
      'role': role,
    };
  }

  AppUser copyWith({String? nickname}) {
    return AppUser(
      id: id,
      phone: phone,
      nickname: nickname ?? this.nickname,
      role: role,
    );
  }
}

class RegistrationResult {
  const RegistrationResult({
    required this.user,
    required this.status,
    required this.message,
  });

  final AppUser user;
  final String status;
  final String message;

  factory RegistrationResult.fromJson(Map<String, dynamic> json) {
    return RegistrationResult(
      user: AppUser.fromJson(json['user'] as Map<String, dynamic>),
      status: json['status'] as String,
      message: json['message'] as String,
    );
  }
}

class AuthSession {
  const AuthSession({
    required this.accessToken,
    required this.user,
    required this.expiresAt,
  });

  final String accessToken;
  final AppUser user;
  final DateTime expiresAt;

  factory AuthSession.fromJson(Map<String, dynamic> json) {
    return AuthSession(
      accessToken: json['accessToken'] as String,
      user: AppUser.fromJson(json['user'] as Map<String, dynamic>),
      expiresAt: DateTime.parse(json['expiresAt'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'accessToken': accessToken,
      'user': user.toJson(),
      'expiresAt': expiresAt.toUtc().toIso8601String(),
    };
  }
}

File: backend/flutter_api/auth_models.go
package flutter_api

import "time"

type AuthUser struct {
	ID           string    `gorm:"primaryKey;size:64" json:"id"`
	Phone        string    `gorm:"size:64;uniqueIndex:idx_auth_users_phone" json:"phone"`
	PasswordHash string    `gorm:"size:255;not null" json:"-"`
	Nickname     string    `gorm:"size:50;not null" json:"nickname"`
	Role         string    `gorm:"size:20;not null" json:"role"`
	Status       string    `gorm:"size:20;not null" json:"status"`
	CreatedAt    time.Time `json:"createdAt"`
	UpdatedAt    time.Time `json:"updatedAt"`
}

func (AuthUser) TableName() string { return "users" }

type AuthSession struct {
	ID           string     `gorm:"primaryKey;size:64" json:"id"`
	UserID       string     `gorm:"size:64;index;not null" json:"userId"`
	TokenHash    string     `gorm:"size:64;uniqueIndex:idx_auth_sessions_token_hash;not null" json:"-"`
	DeviceID     string     `gorm:"size:128;not null" json:"deviceId"`
	CreatedAt    time.Time  `json:"createdAt"`
	LastSeenAt   time.Time  `json:"lastSeenAt"`
	ExpiresAt    time.Time  `json:"expiresAt"`
	RevokedAt    *time.Time `json:"revokedAt,omitempty"`
	RevokeReason string     `gorm:"size:30" json:"-"`
}

func (AuthSession) TableName() string { return "user_sessions" }

type PublicUser struct {
	ID       string `json:"id"`
	Phone    string `json:"phone"`
	Nickname string `json:"nickname"`
	Role     string `json:"role"`
}

type RegisterInput struct {
	Phone    string `json:"phone"`
	Password string `json:"password"`
	Nickname string `json:"nickname"`
	DeviceID string `json:"deviceId"`
}

type LoginInput struct {
	Phone    string `json:"phone"`
	Password string `json:"password"`
	DeviceID string `json:"deviceId"`
}

type UpdateProfileInput struct {
	Nickname string `json:"nickname"`
}

type AuthPrincipal struct {
	UserID    string
	SessionID string
	DeviceID  string
}

type AuthSessionResponse struct {
	User        PublicUser `json:"user"`
	AccessToken string     `json:"accessToken"`
	ExpiresAt   time.Time  `json:"expiresAt"`
}

type AuthError struct {
	Code    string `json:"code"`
	Message string `json:"message"`
}
